tiledb/sm/query/readers/result_tile.h

```diff
@@ -85,13 +85,53 @@ class ResultTile {
     TileTuple() = delete;
 
     /** Constructor with var_size and nullable parameters. */
-    TileTuple(bool var_size, bool nullable) {
+    TileTuple(
+        const uint32_t format_version,
+        const ArraySchema& array_schema,
+        const std::string& name,
+        const uint64_t tile_size,
+        const uint64_t tile_persisted_size,
+        const uint64_t tile_var_size,
+        const uint64_t tile_var_persisted_size,
+        const uint64_t tile_validity_size,
+        const uint64_t tile_validity_persisted_size,
+        const bool var_size,
+        const bool nullable)
+        : fixed_tile_(
```

Comment:
I think there's an ideal opportunity here to look at how we can can capture this variation much more explicitly, because there are very few usages of `fixed_tile_`, `var_tile_`, and `validity_tile_`